pkg/ovs/openflow/ofctrl_builder_test.go

```diff
@@ -16,29 +16,2163 @@ package openflow
+				expectedMatchField: &matchField{
+					class: openflow15.OXM_CLASS_OPENFLOW_BASIC,
+					field: openflow15.OXM_FIELD_IN_PORT,
+					value: &openflow15.InPortField{InPort: 2},
+				},
+			},
+		}
+		for _, tc := range testCases {
+			fb := table.BuildFlow(1).MatchInPort(tc.inPort).(*ofFlowBuilder)
+			assert.Equal(t, tc.inPort, fb.ofFlow.Match.InputPort)
+			assert.Equal(t, 1, len(fb.matchers))
+			assert.Equal(t, tc.expectedMatcherStr, fb.matchers[0])
+
+			flowMod := getFlowMod(t, fb.Done())
+			assert.Equal(t, 1, len(flowMod.Match.Fields))
+			checkMatchField(t, tc.expectedMatchField, flowMod.Match.Fields[0])
+		}
+	})
+	t.Run("MatchSrcIP/MatchDstIP", func(t *testing.T) {
+		testCases := []struct {
+			isSrc              bool
+			ip                 net.IP
+			expectedMatcherStr string
+			expectedMatchField *matchField
+		}{
+			{
+				isSrc:              true,
+				ip:                 net.ParseIP("1.1.1.1"),
+				expectedMatcherStr: "nw_src=1.1.1.1",
+				expectedMatchField: &matchField{
+					class: openflow15.OXM_CLASS_OPENFLOW_BASIC,
+					field: openflow15.OXM_FIELD_IPV4_SRC,
+					value: &openflow15.Ipv4SrcField{Ipv4Src: net.ParseIP("1.1.1.1")},
+				},
+			},
+			{
+				isSrc:              false,
+				ip:                 net.ParseIP("1.1.1.1"),
+				expectedMatcherStr: "nw_dst=1.1.1.1",
+				expectedMatchField: &matchField{
+					class: openflow15.OXM_CLASS_OPENFLOW_BASIC,
+					field: openflow15.OXM_FIELD_IPV4_DST,
+					value: &openflow15.Ipv4DstField{Ipv4Dst: net.ParseIP("1.1.1.1")},
+				},
+			},
+			{
+				isSrc:              true,
+				ip:                 net.ParseIP("fec0::1111"),
+				expectedMatcherStr: "ipv6_src=fec0::1111",
+				expectedMatchField: &matchField{
+					class: openflow15.OXM_CLASS_OPENFLOW_BASIC,
+					field: openflow15.OXM_FIELD_IPV6_SRC,
+					value: &openflow15.Ipv6SrcField{Ipv6Src: net.ParseIP("fec0::1111")},
+				},
+			},
+			{
+				isSrc:              false,
+				ip:                 net.ParseIP("fec0::1111"),
+				expectedMatcherStr: "ipv6_dst=fec0::1111",
+				expectedMatchField: &matchField{
+					class: openflow15.OXM_CLASS_OPENFLOW_BASIC,
+					field: openflow15.OXM_FIELD_IPV6_DST,
+					value: &openflow15.Ipv6DstField{Ipv6Dst: net.ParseIP("fec0::1111")},
+				},
+			},
+		}
+		for _, tc := range testCases {
+			var fb *ofFlowBuilder
+			if tc.isSrc {
+				fb = table.BuildFlow(1).MatchSrcIP(tc.ip).(*ofFlowBuilder)
+				assert.Equal(t, tc.ip, *fb.ofFlow.Match.IpSa)
+			} else {
+				fb = table.BuildFlow(1).MatchDstIP(tc.ip).(*ofFlowBuilder)
+				assert.Equal(t, tc.ip, *fb.ofFlow.Match.IpDa)
+			}
+			assert.Equal(t, 1, len(fb.matchers))
+			assert.Equal(t, tc.expectedMatcherStr, fb.matchers[0])
+
+			flowMod := getFlowMod(t, fb.Done())
+			assert.Equal(t, 1, len(flowMod.Match.Fields))
+			checkMatchField(t, tc.expectedMatchField, flowMod.Match.Fields[0])
+		}
+	})
+	t.Run("MatchSrcIPNet/MatchDstIPNet", func(t *testing.T) {
+		_, ipv4Net1, _ := net.ParseCIDR("1.1.1.0/24")
```

Comment:
Could you add a case for MatchDstIPNet like that the input ipNet is net.IPNet{IP: 1.1.1.100, Mask: 255.255.255.0}. I remember that the final flow realized by OVS is using a match condition nw_dst=1.1.1.0/24 other than  nw_dst=1.1.1.100/24 ?

Reply:
Done